src/main/java/mate/academy/mybookshop/entity/OrderEntity.java

```diff
@@ -0,0 +1,71 @@
+package mate.academy.mybookshop.entity;
+
+import jakarta.persistence.CascadeType;
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.EnumType;
+import jakarta.persistence.Enumerated;
+import jakarta.persistence.FetchType;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.JoinColumn;
+import jakarta.persistence.ManyToOne;
+import jakarta.persistence.OneToMany;
+import jakarta.persistence.Table;
+import jakarta.validation.constraints.Min;
+import jakarta.validation.constraints.NotNull;
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.HashSet;
+import java.util.Set;
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import lombok.ToString;
+import org.hibernate.annotations.SQLDelete;
+import org.hibernate.annotations.SQLRestriction;
+
+@Entity
+@Data
+@SQLDelete(sql = "UPDATE orders SET is_deleted = true WHERE id=?")
+@SQLRestriction(value = "is_deleted=false")
+@Table(name = "orders")
+public class OrderEntity {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @ToString.Exclude
+    @EqualsAndHashCode.Exclude
+    @NotNull
+    @JoinColumn(name = "user_id")
+    @ManyToOne(targetEntity = UserEntity.class, fetch = FetchType.LAZY)
+    private UserEntity user;
+
+    @Enumerated(EnumType.STRING)
+    @Column(nullable = false)
+    private Status status = Status.NEW;
+
+    @NotNull
+    @Min(value = 0)
+    private BigDecimal total;
+
+    @NotNull
+    private LocalDateTime orderDate;
```

Comment:
```suggestion
    @CreationTimestamp
    @NotNull
    private LocalDateTime orderDate;
```
for you to use instead of setting it manually in the service, or you could also add some `@PrePersist` method here and set it there

Reply:
works only without @NotNull